server/controllers/experimentController.js

```diff
@@ -81,24 +82,39 @@ const backfillData = async (req, res, next) => {
   try {
     const result = await experimentsTable.query("SELECT CURRENT_DATE - MIN(date_started) AS date_diff FROM experiments WHERE date_ended IS NULL");
     const dayDiff = result.rows[0]['date_diff'];
-    await backfill(dayDiff);
+    await backfillExposures(dayDiff);
     res.status(200).send("Successfully updated");
   } catch (err) {
     console.log(err);
     res.status(500).send(err.message);
   }
 };
 
-const getAnalysis = async (req, res, next) => {
+const analyzeAll = async (req, res, next) => {
+  try {
+    await runAnalytics();
+    res.status(200).send("Success");
+  } catch (err) {
+    console.log(err);
+    res.status(500).send(err.message);
+  }
+};
+
+const analyzeExperiment = async (req, res, next) => {
   const id = req.params.id;
-  // Statistics and fill in the data in the experiments table
-  res.status(200).send(req.updatedExpt); // use for now until analysis is created
-  // res.status(200).send("Analysis")
+  try {
+    await runAnalytics(id);
+    res.status(200).send("Success");
```

Comment:
If we're running this in the `put` request for an experiment, should we be sending back the experiment data afterwards?

Reply:
Maybe rebase and then re-use some of my code that I created to get the latest metrics and place it on the `updatedExpt` object? Or we could just merge now and do that in a future pull request.

Reply:
oh yeah true, would that just be the mean and p-value stuff?

Reply:
I'm not sure how to do this efficiently, it seems like it's going to do some duplicate work since we're already doing a lot of the metrics work in the `editExperiment` controller. How about if we just remove the `analyzeExperiment` from the PUT endpoint and instead just let the user click a "Refresh" button that'll call the endpoint? We can just show "No data yet" in the meantime before they click Refresh

Reply:
Or another way maybe is to have the frontend call the GET `analyze-experiments/:id` endpoint right after it calls the edit action, in the stop experiment button click handler